services:
  users:
    build: .
    container_name: users
    command: bash -c "python3 ./django/vault/vault_init.py"
    restart: always
    environment:
      - VAULT_DATABASE_NAME=users_db
      - REQUESTS_CA_BUNDLE=/usr/src/app/vault/vault.crt
    volumes:
      - .:/usr/src/app/django
      - ../vault/data:/usr/src/app/vault/
    networks:
      - website-network
      - users-network
    ports:
      - "8002:8002"
    depends_on:
      users_db:
        condition: service_healthy
  users_db:
    build: ./postgre
    container_name: users_db
    restart: always
    volumes:
      - users_db:/var/lib/postgresql/data/
    networks:
      - users-network
      - vault-network
    environment:
      - POSTGRES_DB=${DB_NAME}
      - POSTGRES_USER=${DB_USER}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_ROLE=${DB_ROLE}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d ${DB_NAME} -U ${DB_USER}"]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  users_db:

networks:
  website-network:
    name: website_network
  users-network:
    name: users_network
  vault-network:
    name: vault_network